eth/downloader/whitelist/milestone.go

```diff
@@ -0,0 +1,143 @@
+package whitelist
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type milestone struct {
```

Comment:
is it the same structure as checkoint? if so please check could we use alias or embedding here?

```
// alias
type milestone checkpoint

// embedding
type milestone struct {
    checkpoint
}

Reply:
and remove most of copy-paste below

Reply:
Yes, Same structure. I will check the both approach,

Reply:
To implement this approach, we need to change the name of all the functions and variable we have used for milestone.